### PR TITLE
Using lazy mapper everywhere

### PR DESCRIPTION
**Pull request by sebastianroesch**
Should solve https://github.com/pusher/faros/issues/135.

I had to pass the mapper to the `ReconcileGitTrack` and `ReconcileGitTrackObject` as it was using the default mapper before.

This works for me.

### CODE REVIEW

**Comment by JoelSpeed** [comment]
/ok-to-test

Looks like the underlying issue here is that the `client` we receive from the manager is not updating correctly then? Does this mean that maybe the issue actually lies inside `controller-runtime`?

I've not done a thorough review yet on this, need to have a deeper look as this is a larger change than I had first imagined 🤔

**Comment by sebastianroesch** [comment]
If I understood it correctly, the default client uses the `DiscoveryRESTCMapper`: https://github.com/kubernetes-sigs/controller-runtime/blob/master/pkg/client/client.go#L66

As does the cache that is used for Informers: https://github.com/kubernetes-sigs/controller-runtime/blob/master/pkg/cache/cache.go#L125

**Comment by sebastianroesch** [comment]
I moved the creation of the client and cache to `manager/main` so that the `manager` uses those instances. This fixes the tests too.

File: cmd/manager/main.go [hunk 1]
```diff
@@ -70,6 +92,12 @@ func main() {
 		MetricsBindAddress:      *metricsBindAddress,
 		SyncPeriod:              syncPeriod,
 		Namespace:               farosflags.Namespace,
+		NewClient: func(cache cache.Cache, config *rest.Config, options client.Options) (client.Client, error) {
```
Comment (JoelSpeed) [inline cmd/manager/main.go]: I've been looking through the available options here and it seems we can just provide `utils.NewRestMapper` as the `MapperProvider` and use the in-built utils to construct the clients, WDYT? Would result in a 1 line diff in this file 😄 

https://github.com/kubernetes-sigs/controller-runtime/blob/4276f3895df0acc9249f817eb86a47a3db6b7a9e/pkg/manager/manager.go#L96

**Pull request merged**